include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -305,17 +304,17 @@ namespace alpaka
         //! \tparam TKernelFn Kernel function object type.
         //! \tparam TArgs Kernel function object argument types as a parameter pack.
         template<typename TApi, typename TDev, typename TDim, typename TIdx, typename TKernelFn, typename... TArgs>
-        struct FunctionAttributes<AccGpuUniformCudaHipRt<TApi, TDim, TIdx>, TDev, KernelBundle<TKernelFn, TArgs...>>
+        struct FunctionAttributes<AccGpuUniformCudaHipRt<TApi, TDim, TIdx>, TDev, TKernelFn, TArgs...>
         {
-            //! \param kernelBundle Kernel bundeled with it's arguments. The function attributes of this kernel will be
-            //! determined. Max threads per block is one of the attributes.
-            //! \return KernelFunctionAttributes instance. For GPU backend, all values are set by calling the
-            //! corresponding API functions. The default version always returns an instance with zero fields. For CPU,
-            //! the field of max threads allowed by kernel function for the block is 1.
+            //! \param dev The device instance
+            //! \param kernelFn The kernel function object which should be executed.
+            //! \param args The kernel invocation arguments.
+            //! \return KernelFunctionAttributes instance. The default version always returns an instance with zero
+            //! fields. For CPU, the field of max threads allowed by kernel function for the block is 1.
             ALPAKA_FN_HOST static auto getFunctionAttributes(
-                TDev const&,
-                [[maybe_unused]] KernelBundle<TKernelFn, TArgs...> const& kernelBundle)
-                -> alpaka::KernelFunctionAttributes
+                [[maybe_unused]] TDev const& dev,
+                [[maybe_unused]] TKernelFn const& kernelFn,
+                [[maybe_unused]] TArgs&&... args) -> alpaka::KernelFunctionAttributes
```

Comment:
We do not have the kernel bundle anymore in this PR.
But the behavior is equal to the behavior before, I would say it is implicit casted because it follows the C*+ rules, but I am not 100% sure.

Reply:
The evaluated function attributes should fix because we use the same function signature we use for the kernel call. I do not see that this can be different.